src/management-system-v2/components/confirmation-button.tsx

```diff
@@ -0,0 +1,77 @@
+import { ComponentProps, FC, PropsWithChildren, useState } from 'react';
+import { Button, Modal, Tooltip } from 'antd';
+
+type ConfirmationModalProps = {
+  onConfirm: (...args: any[]) => Promise<any> | any;
```

Comment:
Why does the type allow a function with args if the call to onConfirm does not pass any args.
Couldn't the return value just be `Promise<void> | void` since the return value is not really used for anything?